### ISSUE TITLE
[MEDIUM] DDPM score_model.eval() never restored after sampling

### ISSUE THREAD

**Issue by Maxelee** [Maxelee/vdm_BIND#12]
## Description

In `vdm/ddpm_model.py`, the `sample()` method calls `self.score_model.eval()` but never restores it to `self.score_model.train()` afterward. If sampling is called during training (e.g., for validation visualization), the model stays in eval mode for subsequent training steps.

## Impact

- BatchNorm/Dropout layers stay in eval mode during training
- Training metrics may silently degrade after the first validation sample
- Hard to debug since gradients still flow

## Fix

Use a context manager or explicitly restore:

```python
def sample(self, ...):
    was_training = self.score_model.training
    self.score_model.eval()
    try:
        # ... sampling code ...
    finally:
        if was_training:
            self.score_model.train()
```

## Files
`vdm/ddpm_model.py` — `sample()` method
